libs/pika/execution/tests/include/algorithm_test_utils.hpp

```diff
@@ -96,6 +96,51 @@ struct error_sender
     }
 };
 
+template <typename T>
+struct const_reference_error_sender
+{
+    template <template <class...> class Tuple,
+        template <class...> class Variant>
+    using value_types = Variant<Tuple<T>>;
+
+    template <template <class...> class Variant>
+    using error_types = Variant<std::exception_ptr const&>;
+
+    static constexpr bool sends_done = false;
+
+    using completion_signatures =
+        pika::execution::experimental::completion_signatures<
+            pika::execution::experimental::set_value_t(),
+            pika::execution::experimental::set_error_t(std::exception_ptr&)>;
+
+    template <typename R>
+    struct operation_state
+    {
+        std::decay_t<R> r;
+        friend void tag_invoke(pika::execution::experimental::start_t,
+            operation_state& os) noexcept
+        {
+            try
+            {
+                throw std::runtime_error("error");
+            }
+            catch (...)
+            {
+                auto const& e = std::current_exception();
+                pika::execution::experimental::set_error(
+                    std::move(os.r), e);
+            }
+        }
```

Comment:
I forgot about `make_exception_ptr` (https://en.cppreference.com/w/cpp/error/make_exception_ptr):
```suggestion
            pika::execution::experimental::set_error(
                    std::move(os.r), std::make_exception_ptr(std::runtime_error("error")));
        }
```